### PR TITLE
Allow to explicitly name the feedstock

### PR DESCRIPTION
**Pull request by xhochy**
Specify the generated feedstock name via:

```
extra:
  feedstock_name: explicit_name
```

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by isuruf** [comment]
You'll probably need to fix the linter.

File: conda_smithy/utils.py [hunk 1]
```diff
@@ -12,6 +12,15 @@
 import ruamel.yaml
 
 
+def get_feedstock_name_from_meta(meta):
+    """Resolve the feedtstock name from the parsed meta.yaml."""
+    if "parent_recipe" in meta.meta["extra"]:
+        return meta.meta["extra"]["parent_recipe"]["name"]
+    elif "feedstock_name" in meta.meta["extra"]:
```
Comment (isuruf) [inline conda_smithy/utils.py:19]: Shouldn't this come before? `parent_recipe` is defined for all outputs except for the top level package.

**Comment by xhochy** [comment]
Fixed the linter and fixed the lint.

**Comment by isuruf** [comment]
Have you tried this and checked if it actually works?

**Comment by xhochy** [comment]
I did test both but forgot to rerun the linter afterwards, this is also fixed.

See the output of the manual tests:

```
% conda-smithy init --feedstock-directory ~/tmp/r-clang-activation-feedstock ~/Development/staged-recipes/recipes/r-clang-activation
No azure token. Create a token and
put it in ~/.conda-smithy/azure.token
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Leeres Git-Repository in /Users/uwe/tmp/r-clang-activation-feedstock/.git/ initialisiert
[master (Root-Commit) 3991f03] Initial feedstock commit with conda-smithy 3.7.1.dev20+explicit-feedstock-name.
 11 files changed, 161 insertions(+)
 create mode 100644 conda-forge.yml
 create mode 100644 recipe/LICENSE.txt
 create mode 100644 recipe/activate-r-clang_win-64.bat
 create mode 100644 recipe/bld.bat
 create mode 100644 recipe/conda_build_config.yaml
 create mode 100644 recipe/deactivate-r-clang_win-64.bat
 create mode 100644 recipe/meta.yaml
 create mode 100644 recipe/r_clang_wrapper.bat
 create mode 100644 recipe/r_clangxx_wrapper.bat
 create mode 100644 recipe/run_test.bat
 create mode 100644 recipe/test.cpp

Repository created, please edit conda-forge.yml to configure the upload channels
and afterwards call 'conda smithy register-github'

% conda-smithy register-github --user xhochy ~/tmp/r-clang-activation-feedstock
No azure token. Create a token and
put it in ~/.conda-smithy/azure.token
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
Adding in variants from /Users/uwe/conda_build_config.yaml
Adding in variants from /Users/uwe/tmp/r-clang-activation-feedstock/recipe/conda_build_config.yaml
Created xhochy/r-clang-activation-feedstock on github

Repository registered at github, now call 'conda smithy register-ci'
```

Generated https://github.com/xhochy/r-clang-activation-feedstock

```
% conda-smithy recipe-lint ~/Development/staged-recipes/recipes/r-clang-activation
No azure token. Create a token and
put it in ~/.conda-smithy/azure.token
/Users/uwe/Development/staged-recipes/recipes/r-clang-activation is in fine form
```

**Pull request merged**